addon/components/async-button.js

```diff
@@ -38,9 +38,18 @@ const ButtonComponent = Component.extend(positionalParamsMixin, {
       set(this, 'promise', promise);
     };
 
-    let actionArguments = ['action', callbackHandler, ...params];
+    if (typeof this.attrs.action === 'function') {
+      let promise = this.attrs.action();
```

Comment:
I think this depends on our intended versioning.  If we do not intend to bump to 0.8.0 (signifying breaking changes), we would need to do something like:

```js
let deprecatingCallbackHandler = (promise) => {
  Ember.deprecate(
    'When using closure style actions with ember-async-button please ' +
    'return the promise instead of using the callback argument. The ' + 
    'callback for closure actions will be removed in future versions.',
    promise
  );

  callbackHandler(promise);
};

let promise = this.attrs.action(deprecatingCallbackHandler, ...params);
```

We could release ^^ as a 0.7.1, and then remove that extra work in favor of curried args (which I agree is better) as 0.8.0.